hashintel/hash
# Configure Sentry Replay

## 🌟 What is the purpose of this PR?

This PR updates Sentry and introduces [@sentry/replay](https://www.npmjs.com/package/@sentry/replay). The setup has been generally aligned with [blockprotocol/blockprotocol](https://github.com/blockprotocol/blockprotocol).




## 🔗 Related links





- [Asana task](https://app.asana.com/0/1201095311341924/1203176157000688/f) _(internal)_
- https://github.com/blockprotocol/blockprotocol/pull/682
- https://github.com/blockprotocol/blockprotocol/pull/694


## 🚫 Blocked by




- #1234 

## 🔍 What does this change?

`next.config.js` contains several statements like `process.env.NEXT_PUBLIC_X = process.env.X`. This simplifies infrastructure setup as it makes it easier to use env variables outside Next.js. The PR follows this principle for `SENTRY_REPLAYS_SAMPLING_RATE`. `NEXT_PUBLIC_SENTRY_DSN` has been renamed to `SENTRY_DSN` too and new variables are created on Vercel accordingly.

## 📜 Does this require a change to the docs?



No

## ⚠️ Known issues

- https://github.com/getsentry/sentry-replay/issues/246#issuecomment-1286004492

## 🐾 Next steps

Check 

## 🛡 What tests cover this?

- Manual

## ❓ How to test this?

- Open latest Vercel deployment
- Make sure it does not crash in any new ways
